### PR TITLE
Added type declarations

### PR DESCRIPTION
**Pull request by nilswsr**
I've added type declarations for all the oneDriveAPI functions. The two primary aspects are:

1. Types of the "params" object, which is used for the function parameters
2. Function return types based on [@microsoft/microsoft-graph-types](https://github.com/microsoftgraph/msgraph-typescript-typings)

Furthermore, I deprecated params.rootItemId and params.query, as they are the same as params.itemId and params.queryParamters, which are used in different functions. This provides more consistency.

This time I bumped the package version to `1.1.0`. I think we should abide by the [npm guidelines](https://docs.npmjs.com/about-semantic-versioning) concerning semantic versioning.

There are no breaking changes, but for the typings to work correctly, you will need to:

Convert the import of the module from...
```javascript
const oneDriveAPI = require("onedrive-api")
```
to...
```javascript
import oneDriveAPI from "onedrive-api";
```

And replace the deprecated params.

### CODE REVIEW

**Comment by nilswsr** [comment]
@namdien177 Could you review this please? Thank you

**Review by namdien177 — COMMENTED** [review]
I only have a very small nitpicking comment, but overall, a stellar commit!

File: README.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ npm install onedrive-api
 ### Require module
 
 ```javascript
-const oneDriveAPI = require("onedrive-api");
+import oneDriveAPI from "onedrive-api";
```
[the remark this answers is not in the document]
Comment (nilswsr) [reply README.md:41]: Yes, that would make sense. I changed it.

File: lib/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  type UpdateParams = {
+    toUpdate: Partial<DriveItem>
+  } & AccessToken & DriveSpecification & ItemSpecifiation;
+  type UpdateFn = (params: UpdateParams) => Promise<DriveItem>;
+
+  type GetMetadataParams = AccessToken & DriveSpecification & ItemSpecifiation;
+  type GetMetadataFn = (params: GetMetadataParams) => Promise<DriveItem>;
+
+  type DownloadParams = {
+    format?: "glb" | "html" | "jpg" | "pdf"
```
[the remark this answers is not in the document]
Comment (nilswsr) [reply lib/index.d.ts:46]: I think we should overall go over the README, the JSDoc, ... and smoothen it out. There are some inconsistencies we should get rid of.

But as you say, that's probably for another day ;)

**Review by namdien177 — APPROVED** [review]

**Pull request merged**